### PR TITLE
AudioServer: Assorted infrastructure work

### PR DESCRIPTION
**Pull request by rburchell**
* Add a LibAudio, and move WAV file parsing there (via AWavFile and AWavLoader)
* Add CLocalSocket, and CSocket::connect() variant for local address types.
  We make some small use of this in WindowServer (as that's where we
  modelled it from), but don't get too invasive as this PR is already
  quite large, and the WS I/O is a bit carefully done
* Add an AClientConnection which will eventually be used to talk to
  AudioServer (and make use of it in Piano, though right now it really
  doesn't do anything except connect, using our new CLocalSocket...)

### CODE REVIEW

**Comment by rburchell** [comment]
Sorry for the lack of structure, I sort of barged around on all sides on this without splitting it into smaller pieces. I'll try to do better next time.

File: Applications/Piano/main.cpp [hunk 1]
```diff
@@ -10,6 +11,13 @@ static int s_pipefds[2];
 
 int main(int argc, char** argv)
 {
+    AClientConnection audio_connection;
+    CFile audio("/dev/audio");
```
Comment (rburchell) [inline Applications/Piano/main.cpp]: Oh, hmm, this was a bad merge resolution I guess

File: Libraries/LibAudio/AClientConnection.cpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        m_connection.write("Hello, friends");
+    };
+
+    int retries = 1000;
+    while (retries) {
+        if (m_connection.connect(CSocketAddress::local("/tmp/asportal"))) {
+            break;
+        }
+
+#ifdef GEVENTLOOP_DEBUG
```
Comment (awesomekling) [inline Libraries/LibAudio/AClientConnection.cpp]: `GEVENTLOOP_DEBUG` huh :)

Comment (rburchell) [reply Libraries/LibAudio/AClientConnection.cpp]: Hahah, sorry

**Pull request merged**